### PR TITLE
dataconnect(test): ConnectRPCIntegrationTest.kt: add some more tests

### PR DESCRIPTION
**Pull request by dconeybe**
This PR adds several integration tests to `ConnectRPCIntegrationTest.kt` to verify error handling for duplicate flow collection and duplicate request IDs, while also enhancing the test data generators for Data Connect stream requests.

### Highlights
- **Error Handling Tests**: Added tests to ensure the system correctly handles edge cases, such as multiple collectors on the same response flow and the use of duplicate request IDs.
- **Enhanced Test Generators**: Refactored and expanded the `Arb` (Arbitrary) generators for stream requests to support subscriptions and provide more granular control over query and mutation execution requests.
- **Improved Coverage**: Added a dedicated test for the new `validSubscribeStreamRequestArb` generator to ensure it produces valid and functional subscription requests.

### Changelog
<details>
<summary>1 file affected</summary>

- **ConnectRPCIntegrationTest.kt**: Added `collectingFlowTwiceThrows`, `registeringRequestIdTwiceThrows`, and `testValidSubscribeStreamRequestArb` tests; refactored `validStreamRequestArb` and added specialized `Arb` generators for execute and subscribe requests.
</details>

### CODE REVIEW

File: firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/ConnectRPCIntegrationTest.kt [hunk 1]
```diff
@@ -299,6 +302,68 @@ class ConnectRPCIntegrationTest : DataConnectIntegrationTestBase() {
...
+  @Test
+  fun registeringRequestIdTwiceThrows() = runTest {
+    val streams = connect()
+    streams.sendInitRequest()
+    val requestId = requestIdArb().sample()
+    val (streamRequest1, streamRequest2) =
+      validSubscribeStreamRequestArb(
+          requestId = Arb.constant(requestId),
+          validExecuteQueryRequest =
+            @OptIn(DelicateKotest::class) validExecuteQueryRequestArb().distinct(),
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/ConnectRPCIntegrationTest.kt:346]: Ignoring, because "practically unique" is not the same as "unique". The edge cases of Arb.uuid() could easily generate the same value multiple times.

File: firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/ConnectRPCIntegrationTest.kt [hunk 2]
```diff
@@ -299,6 +302,68 @@ class ConnectRPCIntegrationTest : DataConnectIntegrationTestBase() {
...
+        )
+        .pair()
+        .sample()
+
+    check(streamRequest1.requestId == streamRequest2.requestId)
+    check(streamRequest1.subscribe != streamRequest2.subscribe)
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/ConnectRPCIntegrationTest.kt:352]: Ignoring. These checks are not testing the desired behavior but, rather, asserting internal preconditions for the test to be valid, assertions that should never fail.

File: firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/ConnectRPCIntegrationTest.kt [hunk 3]
```diff
@@ -314,6 +379,30 @@ class ConnectRPCIntegrationTest : DataConnectIntegrationTestBase() {
...
+  fun testValidSubscribeStreamRequestArb() = runTest {
+    val streams = connect()
+    streams.sendInitRequest()
+
+    // Use distinct UUIDs in the "subscribe" requests because the backend appears to do some weird
+    // de-duping if the same query is subscribed to with distinct request IDs.
+    val subscribeStreamRequestArb =
+      validSubscribeStreamRequestArb(
+        validExecuteQueryRequest =
+          validExecuteQueryRequestArb(keyArb(@OptIn(DelicateKotest::class) Arb.uuid().distinct()))
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/ConnectRPCIntegrationTest.kt:392]: Ignoring. As mentioned above, "practically unique" is not sufficient as the edge cases can easily produce duplicates (e.g. a UUID of all zeroes)

File: firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/ConnectRPCIntegrationTest.kt [hunk 4]
```diff
@@ -314,6 +379,30 @@ class ConnectRPCIntegrationTest : DataConnectIntegrationTestBase() {
...
+      )
+
+    streams.incomingResponses.test {
+      checkAll(propTestConfig, subscribeStreamRequestArb) { streamRequest ->
+        check(streamRequest.hasSubscribe())
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply firebase-dataconnect/src/androidTest/kotlin/com/google/firebase/dataconnect/ConnectRPCIntegrationTest.kt:397]: Ignoring. This code is validating an internal precondition of the test, a precondition that should never fail. You are correct, however, that "shouldBe" should be used when validating the result or side effect of the behavior under test; however, the particular line of code in question is _not_ that.

**Review by stephenarosaj — APPROVED** [review]
